File: src/dataset/javascript/sieve_exercism-ThomasZumsteg/openAI/ChatGPT4_sieve_v2.js
class Sieve {
    constructor(size) {
        this.primes = [];
        if (size < 2) return;

        const sieve = new Uint8Array(size + 1);
        sieve.fill(1);
        sieve[0] = sieve[1] = 0;

        for (let n = 2; n * n <= size; n++) {
            if (sieve[n]) {
                for (let m = n * n; m <= size; m += n) {
                    sieve[m] = 0;
                }
            }
        }

        for (let n = 2; n <= size; n++) {
            if (sieve[n]) this.primes.push(n);
        }
    }
}

module.exports = Sieve;

File: src/dataset/python/sieve_exercism-ThomasZumsteg/openAI/ChatGPT4_sieve_v2.py
"""Finds prime numbers"""

def sieve(lim):
    """Implementation of sieve of Eratosthenes"""
    if lim < 2:
        return []
    sieve = [True] * (lim + 1)
    sieve[0] = sieve[1] = False
    primes = []
    for num in range(2, int(lim**0.5) + 1):
        if sieve[num]:
            primes.append(num)
            for multiple in range(num * num, lim + 1, num):
                sieve[multiple] = False
    primes.extend(num for num in range(int(lim**0.5) + 1, lim + 1) if sieve[num])
    return primes